### PR TITLE
chore: move configUtil from core to utils

### PR DESCRIPTION
**Pull request by gbockus-sf**
### What does this PR do?
Moved configUtil from core to utils and update calls/test in core to use the utils version.
Update utils telemetry to use common configUtils.

### What issues does this PR fix or reference?
@W-11598416@

### Functionality Before
core used configUtils in core.
utils telemetry had it's own configUtils

### Functionality After
core & utils telemetry use common configUtils.

### CODE REVIEW

File: packages/salesforcedx-utils-vscode/src/config/configUtil.ts [hunk 1]
```diff
@@ -1,90 +1,117 @@
...
-import * as path from 'path';
 import { ConfigAggregatorProvider } from '../providers';
-import { TelemetryService } from '../telemetry/telemetry';
-import { getRootWorkspacePath } from '../workspaces';
 
 export enum ConfigSource {
   Local,
   Global,
   None
 }
```
Comment (gbockus-sf) [inline packages/salesforcedx-utils-vscode/src/config/configUtil.ts:22]: This was basically combining the new configUtils in core with the new configUtils in vscode-utils.

File: packages/salesforcedx-utils-vscode/test/unit/telemetry/cliConfiguration.test.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { GlobalCliEnvironment } from '../../../src/cli';
+import { ConfigUtil } from '../../../src/config/configUtil';
+import {
+  disableCLITelemetry,
+  ENV_SFDX_DISABLE_TELEMETRY,
+  isCLITelemetryAllowed
+} from '../../../src/telemetry/cliConfiguration';
```
Comment (gbockus-sf) [inline packages/salesforcedx-utils-vscode/test/unit/telemetry/cliConfiguration.test.ts:7]: new jest style test for cliConfiguration.

File: packages/salesforcedx-utils-vscode/src/telemetry/cliConfiguration.ts [hunk 3]
```diff
@@ -20,9 +19,8 @@ export function disableCLITelemetry() {
 
 export async function isCLITelemetryAllowed(): Promise<boolean> {
   try {
-    const disabledConfig =
-      (await ConfigUtil.getConfigValue(SFDX_CONFIG_DISABLE_TELEMETRY)) || '';
-    return disabledConfig !== 'true';
+    const disabledConfig = await ConfigUtil.isTelemetryDisabled();
+    return !disabledConfig;
```
Comment (klewis-sfdc) [inline packages/salesforcedx-utils-vscode/src/telemetry/cliConfiguration.ts:23]: What do you think about changing the name of this variable from 'disabledConfig' to something like 'isTelemetryDisabled'?

File: packages/salesforcedx-vscode-core/src/util/index.ts [hunk 4]
```diff
@@ -13,7 +13,6 @@ export {
   isSFDXContainerMode,
   showCLINotInstalledMessage
 } from './cliConfiguration';
-export { ConfigSource, ConfigUtil } from './configUtil';
```
Comment (klewis-sfdc) [inline packages/salesforcedx-vscode-core/src/util/index.ts:16]: 👍🏽

File: packages/salesforcedx-utils-vscode/src/auth/authUtil.ts [hunk 5]
```diff
@@ -24,9 +24,7 @@ export class AuthUtil {
     enableWarning: boolean
   ): Promise<string | undefined> {
     try {
-      const defaultUserName = await ConfigUtil.getConfigValue(
-        DEFAULT_USERNAME_KEY
-      );
+      const defaultUserName = await ConfigUtil.getDefaultUsernameOrAlias();
```
Comment (klewis-sfdc) [inline packages/salesforcedx-utils-vscode/src/auth/authUtil.ts:27]: Should be able to remove the default_username_key import up above on line 11 now

File: packages/salesforcedx-utils-vscode/test/unit/telemetry/cliConfiguration.test.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
+import { GlobalCliEnvironment } from '../../../src/cli';
```
Comment (klewis-sfdc) [inline packages/salesforcedx-utils-vscode/test/unit/telemetry/cliConfiguration.test.ts:1]: Should we add a copyright statement to the top of this file?

File: packages/salesforcedx-utils-vscode/src/auth/authUtil.ts (hunk 5 above)
Comment (gbockus-sf) [reply packages/salesforcedx-utils-vscode/src/auth/authUtil.ts:27]: good eye

File: packages/salesforcedx-utils-vscode/src/telemetry/cliConfiguration.ts (hunk 3 above)
Comment (gbockus-sf) [reply packages/salesforcedx-utils-vscode/src/telemetry/cliConfiguration.ts:23]: 👍 works for me

File: packages/salesforcedx-utils-vscode/test/unit/telemetry/cliConfiguration.test.ts (hunk 6 above)
Comment (gbockus-sf) [reply packages/salesforcedx-utils-vscode/test/unit/telemetry/cliConfiguration.test.ts:1]: 👍

**Review by klewis-sfdc — APPROVED** [review]
LGTM!  Thanks, Gordon!

File: packages/salesforcedx-test-utils-vscode/src/testrunner.ts [hunk 7]
```diff
@@ -30,29 +30,34 @@ let mocha = new Mocha({
   reporter: 'mocha-multi-reporters'
 });
 
-function configure(mochaOpts: any): void {
+function configure(mochaOpts: any, xmlOutputDirectory: string): void {
```
Comment (gbockus-sf) [inline packages/salesforcedx-test-utils-vscode/src/testrunner.ts:33]: Update to pass the directory where the xml reports should be written.

File: packages/salesforcedx-test-utils-vscode/src/testrunner.ts [hunk 8]
```diff
@@ -30,29 +30,34 @@ let mocha = new Mocha({
...
+    );
+    const xunitOutputFileLocation = paths.join(
+      xmlOutputDirectory,
+      'xunit-vscodeIntegrationTests.xml'
+    );
+
+    console.log('Output locations for reporters: ', {
+      junitOutputFileLocation,
+      xunitOutputFileLocation
+    });
```
Comment (gbockus-sf) [inline packages/salesforcedx-test-utils-vscode/src/testrunner.ts:51]: Log it out so this is less timely in figuring out next time we run into a related issue.

File: packages/salesforcedx-vscode-apex-debugger/test/vscode-integration/index.ts [hunk 9]
```diff
@@ -7,13 +7,16 @@
...
-  useColors: true, // colored output from test results
-  timeout: 360000
-});
+testRunner.configure(
+  {
+    ui: 'bdd', // the TDD UI is being used in extension.test.ts (suite, test, etc.)
+    useColors: true, // colored output from test results
+    timeout: 360000
+  },
+  normalize(join(__dirname, '..', '..', '..'))
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-apex-debugger/test/vscode-integration/index.ts:19]: update all the places where we use the testRunner to pass the correct directory for writing the xml results.  Note this is three levels back from the current directory which is out/test/vscode-integration

**Pull request merged**